src/main/kotlin/no/nav/syfo/application/VedtakService.kt

```diff
@@ -66,4 +67,22 @@ class VedtakService(
             }
         }
     }
+
+    fun sendVedtakVarsel(
+        personident: Personident,
+        vedtak: Vedtak
+    ): Result<Vedtak> {
+        val journalpostId = vedtak.journalpostId
+            ?: return Result.failure(Exception("JournalpostId is missing for vedtak ${vedtak.uuid}"))
+
+        esyfovarselHendelseProducer.sendVedtakVarsel(
+            personident = personident,
+            journalpostId = journalpostId,
+            vedtak = vedtak
+        )
+
+        // TODO: Update varsel_published_at column for vedtak
+
+        return Result.success(vedtak)
```

Comment:
Kunne dette vært hakket cleanere hvis det som blir returnert er resultatet av `esyfovarselHendelseProducer` 😊 Hva tenker du? Da lurer jeg også på om man slipper å wrappe `vedtak` i `Result.sucess(…)`🤔

Reply:
Japp, så litt på det. For når vi legger til update-greia på domeneobjektet her på linje 84, vil vi da returnere resultatet fra produseren, eller det oppdaterte vedtaket fra funksjonen? 🤔